pkg/virt-api/webhooks/mutating-webhook/mutators/vm-mutator_test.go

```diff
@@ -350,6 +367,829 @@ var _ = Describe("VirtualMachine Mutator", func() {
 		}
 	})
 
+	Context("with InferFromVolume enabled", func() {
+
+		var (
+			pvc               *k8sv1.PersistentVolumeClaim
+			dvWithSourcePVC   *v1beta1.DataVolume
+			dvWithAnnotations *v1beta1.DataVolume
+			dsWithSourcePVC   *v1beta1.DataSource
+			dsWithAnnotations *v1beta1.DataSource
+		)
+
+		const (
+			inferVolumeName     = "inferVolumeName"
+			defaultInferedName  = "defaultInferedName"
+			defaultInferedKind  = "defaultInferedKind"
+			pvcName             = "pvcName"
+			dsWithSourcePVCName = "dsWithSourcePVC"
+			dsWithAnnotionsName = "dsWithAnnotionsName"
+			unknownPVCName      = "unknownPVC"
+			unknownDVName       = "unknownDV"
+		)
+
+		BeforeEach(func() {
+			pvc = &k8sv1.PersistentVolumeClaim{
+				ObjectMeta: k8smetav1.ObjectMeta{
+					Name:      pvcName,
+					Namespace: vm.Namespace,
+					Annotations: map[string]string{
+						instancetype.DefaultInstancetypeAnnotation:     defaultInferedName,
+						instancetype.DefaultInstancetypeKindAnnotation: defaultInferedKind,
+						instancetype.DefaultPreferenceAnnotation:       defaultInferedName,
+						instancetype.DefaultPreferenceKindAnnotation:   defaultInferedKind,
+					},
+				},
+				TypeMeta: k8smetav1.TypeMeta{
```

Comment:
Is `TypeMeta` actually required, or will the client fill it in?

Reply:
I thought the fakeclients didn't add this in but I've removed it and the tests still pass, thanks.